content/en/tracing/trace_collection/custom_instrumentation/go/migration.md

```diff
@@ -0,0 +1,51 @@
+---
+title: Migrating from v1 to v2 of the Go Tracer
+description: 'Upgrade your Go tracer from v1 to v2.'
+aliases:
+  - /tracing/trace_collection/custom_instrumentation/opentracing/go
+further_reading:
+    - link: 'tracing/trace_collection/custom_instrumentation/go/dd-api'
+      tag: 'Documentation'
+      text: 'Get started with v1 of the Go Tracer'
+---
+
+<div class="alert alert-info">This documentation assumes that you are using version v1.x of the Go tracer. If you are already using v2.x, see <a href="/tracing/trace_collection/custom_instrumentation/go/dd-api">Go Custom Instrumentation using the Datadog API</a> instead.</div>
+
+This guide explains how to migrate from Go Tracer v1.x to v2. **Note**: v2 is in Preview.
+
+Version 2 of the Go tracer introduces significant API improvements:
+
+- Removes interfaces to enable future flexibility
+- Isolates integrations to prevent false positives from security scanners
+- Enforces library patterns to prevent misuse
+
+Following the full release of v2, versions v1.x will receive decreased support. Bug fixes, security fixes, and basic maintenance remains supported in v1 until the end of 2025. After this point, maintenace for v1 is dropped. To receive new features, upgrade your Go Tracer to v2.
```

Comment:
This may be best as a table on the [Go compatibility page](https://docs.datadoghq.com/tracing/trace_collection/compatibility/go). Something like the below, but with actual dates

| Version  | Preview | General Availability (GA) | Maintenance | End-of-life (EOL) |
| - | - | - | - | - |
| v1 | 2022-01-01 | 2022-01-01 | 2024-12-31 | 2026-01-01 |
| v2 | 2024-08-10 | 2025-01-01 | TBD | TBD |

| **Level**                                              | **Support provided**                                                                                                                                |
|--------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------|
| <span id="support-unsupported">Unsupported</span>      |  No implementation. Contact [Datadog support][2] for special requests.                                                                              |
| <span id="support-beta">Preview</span>                 |  Initial implementation. May not yet contain all features. Support for new features and bug and security fixes are provided on a best-effort basis. |
| <span id="support-ga">General Availability (GA)</span> |  Full implementation of all features. Full support for new features and bug and security fixes.                                                     |
| <span id="support-maintenance">Maintenance</span>      |  Full implementation of existing features. Does not receive new features. Support for bug and security fixes only.                                  |
| <span id="support-eol">End-of-life (EOL)</span>        |  No support.                                                                                                                                        |